librz/arch/fcn.c

```diff
@@ -698,11 +694,12 @@ static RzAnalysisBBEndCause run_basic_block_analysis(RzAnalysisTaskItem *item, R
 		}
 	}
 	if ((maxlen - (addrbytes * idx)) > MAX_SCAN_SIZE) {
+		// XXX idx is always 0 here, and maxlen comes from amalysis.bb.maxsize. This makes no sense.
 		RZ_LOG_DEBUG("Skipping large memory region during basic block analysis.\n");
 		maxlen = 0;
 	}
```

Comment:
this looks like remnants of old code. maybe it could be deleted?

Reply:
As pointed out by Rot127 whole thing is going to be rewritten soon. And there are definitely  a bunch of weird stuff. Can't fix it all in this PR. 

The focus this time was on parts that have direct negative effect by producing bad instruction positions which breaks navigation.